src/openms/source/FORMAT/OMSFile.cpp

```diff
@@ -0,0 +1,2675 @@
+    if (!query.exec())
+    {
+      raiseDBError_(query.lastError(), __LINE__, OPENMS_PRETTY_FUNCTION,
+                    "error inserting data");
+    }
+  }
+
+
+  void OMSFile::OMSFileStore::createTableMoleculeType_()
+  {
+    createTable_("ID_MoleculeType",
+                 "id INTEGER PRIMARY KEY NOT NULL, "    \
+                 "molecule_type TEXT UNIQUE NOT NULL");
+    QString sql_insert =
+      "INSERT INTO ID_MoleculeType VALUES "     \
+      "(1, 'PROTEIN'), "                        \
+      "(2, 'COMPOUND'), "                       \
+      "(3, 'RNA')";
+    QSqlQuery query(QSqlDatabase::database(db_name_));
+    if (!query.exec(sql_insert))
+    {
+      raiseDBError_(query.lastError(), __LINE__, OPENMS_PRETTY_FUNCTION,
+                    "error inserting data");
+    }
+  }
+
+
+  void OMSFile::OMSFileStore::createTableDataValue_()
+  {
+    createTable_("DataValue_DataType",
+                 "id INTEGER PRIMARY KEY NOT NULL, "  \
+                 "data_type TEXT UNIQUE NOT NULL");
+    QString sql_insert =
+      "INSERT INTO DataValue_DataType VALUES " \
+      "(1, 'STRING_VALUE'), "                  \
+      "(2, 'INT_VALUE'), "                     \
+      "(3, 'DOUBLE_VALUE'), "                  \
+      "(4, 'STRING_LIST'), "                   \
+      "(5, 'INT_LIST'), "                      \
+      "(6, 'DOUBLE_LIST')";
+    QSqlQuery query(QSqlDatabase::database(db_name_));
+    if (!query.exec(sql_insert))
+    {
+      raiseDBError_(query.lastError(), __LINE__, OPENMS_PRETTY_FUNCTION,
+                    "error inserting data");
+    }
+    createTable_(
+      "DataValue",
+      "id INTEGER PRIMARY KEY NOT NULL, "                               \
+      "data_type_id INTEGER, "                                          \
+      "value TEXT, "                                                    \
+      "FOREIGN KEY (data_type_id) REFERENCES DataValue_DataType (id)");
+    // @TODO: add support for units
+  }
+
+
+  OMSFile::Key OMSFile::OMSFileStore::storeDataValue_(const DataValue& value)
+  {
+    // this assumes the "DataValue" table exists already!
+    // @TODO: split this up and make several tables for different types?
+    QSqlQuery query(QSqlDatabase::database(db_name_));
+    query.prepare("INSERT INTO DataValue VALUES (" \
+                  "NULL, "                         \
+                  ":data_type, "                   \
+                  ":value)");
+    // @TODO: cache the prepared query between function calls somehow?
+    if (!value.isEmpty()) // use NULL as the type for empty values
+    {
+      query.bindValue(":data_type", int(value.valueType()) + 1);
+    }
+    query.bindValue(":value", value.toQString());
+    if (!query.exec())
+    {
+      raiseDBError_(query.lastError(), __LINE__, OPENMS_PRETTY_FUNCTION,
+                    "error inserting data");
+    }
+    return query.lastInsertId().toLongLong();
```

Comment:
@hendrikweisser do I see this correctly, this basically executes an SQL query for every single `DataValue`, isnt that a bit inefficient? Could you also do multiple binds and execute one query to enter many data values into the file? Have you looked at the performance of this?